### ISSUE TITLE
SAC gradients weighted inconsistently

### ISSUE THREAD

**Issue by hartikainen** [rail-berkeley/softlearning#141]
> Also, I'm reading this again and isn't there supposed to be a tf.nn.compute_average_loss before calculating and applying the gradient?

> https://github.com/rail-berkeley/softlearning/blob/84d7589fd5852aff9aa46debda6de39acaec2e0b/softlearning/algorithms/sac.py#L214-L233

_Originally posted by @externalhardrive in https://github.com/rail-berkeley/softlearning/issues/136#issuecomment-620394025_
